backend/src/main/java/gov/cdc/usds/simplereport/idp/authentication/LiveOktaAuthentication.java

```diff
@@ -177,30 +219,224 @@ public String enrollVoiceCallMfa(String userId, String phoneNumber)
       user.enrollFactor(callFactor);
       return callFactor.getId();
     } catch (ResourceException e) {
+      if (e.getStatus() == HttpStatus.BAD_REQUEST.value()) {
+        throw new BadRequestException(e.getError().getMessage(), e);
+      }
       throw new OktaAuthenticationFailureException("Error setting voice call MFA", e);
     }
   }
 
   /**
-   * Using the Okta Management SDK, enroll a user in email MFA. If successful, this enrollment
-   * triggers an activation email to the user with an OTP.
+   * Using the Okta Management SDK, enroll a user in email MFA using their attached profile email.
+   * If successful, this enrollment triggers an activation email to the user with an TOTP.
    *
    * @param userId the user id of the user making the enrollment request.
-   * @param email the user-provided email address to enroll. (note: should we do any validation that
-   *     this email === the enrolled email?)
    * @throws OktaAuthenticationFailureException if the email is invalid or Okta cannot enroll it as
    *     an MFA option.
    */
-  public String enrollEmailMfa(String userId, String email)
-      throws OktaAuthenticationFailureException {
+  public String enrollEmailMfa(String userId) throws OktaAuthenticationFailureException {
     try {
       EmailUserFactor emailFactor = _client.instantiate(EmailUserFactor.class);
-      emailFactor.getProfile().setEmail(email);
       User user = _client.getUser(userId);
+      String userEmail = user.getProfile().getEmail();
+      emailFactor.getProfile().setEmail(userEmail);
       user.enrollFactor(emailFactor);
       return emailFactor.getId();
     } catch (ResourceException e) {
       throw new OktaAuthenticationFailureException("Error setting email MFA", e);
     }
   }
+
+  /**
+   * Using the Okta management SDK, enroll a user in an authentication app for MFA. If successful,
+   * this method returns the factor id and a qr code. The qr code will be passed to the user for
+   * them to finish setting enrolling in-app.
+   *
+   * <p>https://developer.okta.com/docs/reference/api/factors/#response-example-12
+   *
+   * @param userId the user id of the user making the enrollment request.
+   * @param appType the appType of the app being enrolled (for now, one of Okta Verify or Google
+   *     Authenticator.)
+   * @throws OktaAuthenticationFailureException if the app type is not recognized, Okta fails to
+   *     enroll the MFA option, or the result from Okta does not contain a QR code.
+   */
+  public JSONObject enrollAuthenticatorAppMfa(String userId, String appType)
```

Comment:
I'm changing this one to FactorAndQrCode, which is just a POJO with the factor and qr code. 

For the enrollSecurityKey request, I know @nickclyde had formatted the frontend so that it pretty much just accepted the activation JS object from Okta (following [this guide](https://developer.okta.com/docs/reference/api/factors/#get-registration-information-from-webauthn-authenticator-by-calling-the-webauthn-javascript-api)). So I could change the response type to a Java object that just has all the required information, but I don't know how difficult that would be to unwrap on the frontend.